src/redux/actions/articleActions.js

```diff
@@ -76,7 +103,24 @@ export const createArticleOnComment = (slug, comment) => async (dispatch) => {
   }
 };
 
-export const getArticles = () => ({
-  type: GET_ARTICLES,
-  payload: { articles, loading: false },
-});
+
+export const likeArticle = (slug, type, author, article, userId) => async (dispatch) => {
+  try {
+    await API_SERVICE.post(`/articles/${slug}/${type}`);
+    const data = {
+      userId,
+      articleId: article.id,
+      authorId: author.id,
+    };
+    const fetchedArticle = await API_SERVICE.get(`/articles/${slug}`);
+    const articleStatRes = await API_SERVICE.post('/articles/getcurrentarticlestat', data);
```

Comment:
I suggested that since the reaction state is only updated by `fetchedArticle`, move this `articleStatRes` request after you have dispatched `setArticleReaction(reactions)`. This may reduce the lag in applying reactions on the UI

Reply:
Waiting until the two requests are complete may be the reason for that delay